src/lib/error.ts

```diff
@@ -29,6 +29,9 @@ export class UnknownError extends BaseError {
 
   constructor(public error: Error) {
     super();
+    // We need to set the prototype explicitly
+    Object.setPrototypeOf(this, UnknownError.prototype);
```

Comment:
I don't think setting the prototype is necessary for subclasses, only the one directly inheriting from `Error`. I might be mistaken though.

Reply:
From https://github.com/Microsoft/TypeScript-wiki/blob/main/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work 
> As a recommendation, you can manually adjust the prototype immediately after any `super(...)` calls.

> However, any subclass of `FooError` will have to manually set the prototype as well.

I'm interpreting that as meaning we **do** need to set the prototype of the subclasses, what do you think?

Reply:
Agreed. Reading it again, this may not be necessary depending on the emit target, but lets keep it to be safe.